src/main/scala/org/apache/spark/sql/redis/BinaryRedisPersistence.scala

```diff
@@ -34,6 +35,10 @@ class BinaryRedisPersistence extends RedisPersistence[Array[Byte]] {
   override def decodeRow(keyMap: (String, String), value: Array[Byte], schema: StructType,
                          requiredColumns: Seq[String]): Row = {
     val valuesArray: Array[Any] = SerializationUtils.deserialize(value)
-    new GenericRowWithSchema(valuesArray, schema)
+    // Aligning column positions with what Catalyst expecting
+    val alignedSchema = SparkUtils.alignSchemaWithCatalyst(schema, requiredColumns)
```

Comment:
We can optimize it by creating `alignedSchema` once for all rows. To achieve it I guess we will need to change the `decodeRow()` signature by adding a new parameter, it will have two schemas: `requiredSchema` and `persistedSchema`.